### PR TITLE
Torch mlir visualizer

### PR DESCRIPTION
**Pull request by jcamacho1234**
Merge all Visualizer code with main branch here.

### CODE REVIEW

**Review by JRUMacDev — COMMENTED** [review]
The code changes seem good to me. One question - why are the two tests in test/Conversion/ATenToXTen showing as additions? There two tests were already upstreamed. Perhaps I am just misunderstanding what the PR is showing?

**Review by ljfitz — CHANGES_REQUESTED** [review]
Given the verbosity of the target schema, the code looks well put together. I've added a few specific comments, pretty much all style and readability issues.

Most mlir upsteam projects require the use of `clang-format` and you should consider it too. `clang-format -i lib/Transform/ATenVisualGraph.cpp` will format it in place for you.

 I do think it misses a test that checks, for the smallest possible example, the full output. Then the other tests can focus on the parts that are unique to them. From the name and `CHECK` I understood the purpose of most tests, but not `test/Transform/ATenVisualGraph/tiny_yolo_v2_block.mlir`.

**Review by ljfitz — CHANGES_REQUESTED** [review]
Thanks for the changes @jcamacho1234. I was very happy that most of my comments could be marked as `Resolved`. The only one of significance remaining is to improve `test/Transform/ATenVisualGraph/tiny_yolo_v2_block.mlir` to make it clear what it is testing, either by providing a more thorough set of `CHECK` lines or add a comment to explain the key aspect that the single `CHECK-LABEL` covers.

File: test/Transform/ATenVisualGraph/tiny_yolo_v2_block.mlir [hunk 1]
```diff
@@ -10,7 +10,23 @@
 
 // RUN: aten-opt %s -aten-visual-graph='operators-supported-path=%S/../../../lib/Transform/operators_supported.json' | FileCheck %s
 // CHECK-LABEL:     "{{.*}}": {
-//
+// CHECK-LABEL:     "connections"
+// CHECK-LABEL:     {
```
Comment (ljfitz) [inline test/Transform/ATenVisualGraph/tiny_yolo_v2_block.mlir:14]: the use of https://llvm.org/docs/CommandGuide/FileCheck.html#the-check-label-directive everywhere looks odd and that you really just want `CHECK:`. Could you check?

**Review by ljfitz — CHANGES_REQUESTED** [review]

**Review by ljfitz — APPROVED** [review]

**Comment by ljfitz** [comment]
I understand that this was effectively merged via upstream.

**Pull request closed without merging**